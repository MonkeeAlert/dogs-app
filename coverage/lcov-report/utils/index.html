
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for utils</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../index.html">All files</a> utils</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">88.73% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>126/142</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">80.59% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>54/67</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">90.47% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>19/21</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">88.23% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>120/136</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input oninput="onInput()" type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line high'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="constants.ts"><a href="constants.ts.html">constants.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="8" class="abs high">8/8</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="8" class="abs high">8/8</td>
	</tr>

<tr>
	<td class="file high" data-value="functions.ts"><a href="functions.ts.html">functions.ts</a></td>
	<td data-value="92.62" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 92%"></div><div class="cover-empty" style="width: 8%"></div></div>
	</td>
	<td data-value="92.62" class="pct high">92.62%</td>
	<td data-value="122" class="abs high">113/122</td>
	<td data-value="85.71" class="pct high">85.71%</td>
	<td data-value="63" class="abs high">54/63</td>
	<td data-value="94.73" class="pct high">94.73%</td>
	<td data-value="19" class="abs high">18/19</td>
	<td data-value="92.24" class="pct high">92.24%</td>
	<td data-value="116" class="abs high">107/116</td>
	</tr>

<tr>
	<td class="file low" data-value="tinyUrlHelper.ts"><a href="tinyUrlHelper.ts.html">tinyUrlHelper.ts</a></td>
	<td data-value="41.66" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 41%"></div><div class="cover-empty" style="width: 59%"></div></div>
	</td>
	<td data-value="41.66" class="pct low">41.66%</td>
	<td data-value="12" class="abs low">5/12</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="41.66" class="pct low">41.66%</td>
	<td data-value="12" class="abs low">5/12</td>
	</tr>

<tr>
	<td class="file empty" data-value="types.ts"><a href="types.ts.html">types.ts</a></td>
	<td data-value="0" class="pic empty">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at Fri Apr 08 2022 23:23:18 GMT+0300 (Eastern European Summer Time)
            </div>
        <script src="../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../sorter.js"></script>
        <script src="../block-navigation.js"></script>
    </body>
</html>
    