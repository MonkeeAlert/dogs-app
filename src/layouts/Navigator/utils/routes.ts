export type RootStackParamList = {
  CatalogTabs: {
    search?: string;
    refresh?: boolean;
  };
  Search: {
    search?: string;
  };
  User: {};
  Gallery: {
    uri: string;
    img: string;
    size: {
      w: number;
      h: number;
    };
    isConnected?: string;
    search?: string;
  };
  Login: {};
};
