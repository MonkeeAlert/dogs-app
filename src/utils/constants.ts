export const colors = {
  white: '#fffafb',
  lightGray: '#E3F1EA',
  turquoise: '#2BD9A5',
  darkTurquoise: '#2E8A6E',
  darkGray: '#2b2c28',
  black: '#131515',
};

export const text = {
  xs: 7,
  s: 14,
  m: 21,
  l: 28,
  xl: 35,
  xxl: 42,
};

export const animationConfig = {
  duration: 200,
};

export const turquoiseGradientArray = [colors.turquoise, '#2bd9a5aa'];
