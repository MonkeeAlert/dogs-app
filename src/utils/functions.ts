export const flatTree = (o: any): string[] => {
  const _o = Object.assign({}, o);
  const keys = Object.keys(_o);

  for (let k in _o) {
    if (_o[k].length > 0) {
      let a = _o[k].map((i: string) => `${k}-${i}`);

      let idx = keys.indexOf(k);

      keys.splice(idx, 1, ...a);
    }
  }

  return keys;
};

export const findBreedInList = (list: string[], breed: string) => {
  const _arr = JSON.parse(JSON.stringify(list));
  const splitted = breed.toLowerCase().split(' ');

  if (
    _arr.indexOf(splitted.join('-')) !== -1 ||
    _arr.indexOf(splitted.reverse().join('-')) !== -1
  ) {
    const normal = _arr.indexOf(splitted.join('-'));
    const reversed = _arr.indexOf(splitted.reverse().join('-'));

    return normal !== -1 ? _arr[normal] : _arr[reversed];
  } else {
    const res = _arr.filter(w => {
      return splitted.filter(i => w.match(i)).length > 0;
    });

    return res.length === 0 ? [] : res.length === 1 ? res[0] : breed;
  }
};

export const parseImage = (s: string) => {
  const fromBreed = s.slice(s.indexOf('breeds'));
  const arr = fromBreed.split('/');

  return `${arr[1]}-${arr[2]}`;
};
