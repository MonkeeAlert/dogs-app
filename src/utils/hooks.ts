import {Platform} from 'react-native';
import {useSafeAreaInsets} from 'react-native-safe-area-context';

const HEADER_EXPANDED_HEIGHT = 78 + 14;

export const useComputedHeaderValue = () => {
  const insets = useSafeAreaInsets();
  const headerAddedValue = insets.top + 7 * 2;

  const height = Platform.select({
    ios: HEADER_EXPANDED_HEIGHT + headerAddedValue,
    android: HEADER_EXPANDED_HEIGHT + insets.top + headerAddedValue,
    default: HEADER_EXPANDED_HEIGHT + headerAddedValue,
  });

  return {height};
};
