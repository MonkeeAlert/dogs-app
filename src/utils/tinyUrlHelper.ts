import axios from 'axios';
import firestore from '@react-native-firebase/firestore';

const instance = axios.create();

instance.defaults.baseURL = 'https://api.tinyurl.com';
instance.defaults.headers.post['Content-Type'] = 'application/json';

export const createTinyUrl = async (url: string) => {
  const fcData = await firestore().collection('api_keys').doc('tiny_url').get();
  const {key} = fcData.data() as any;

  if (!key && typeof key !== 'string') {
    throw new Error('key was received');
  } else {
    instance.defaults.headers.common['Authorization'] = `Bearer ${key}`;

    const {data} = await instance
      .post(`${instance.defaults.baseURL}/create`, {
        url,
        domain: 'rotf.lol',
      })
      .catch(err => {
        throw new Error(err);
      });

    return data.data.tiny_url;
  }
};
