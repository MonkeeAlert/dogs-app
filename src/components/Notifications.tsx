import {
  View,
  Text,
  StyleSheet,
  Dimensions,
  TouchableOpacity,
} from 'react-native';
import React, {useImperativeHandle, useRef, useState} from 'react';
import {colors, springConfig} from '../utils/constants';
import {useSafeAreaInsets} from 'react-native-safe-area-context';
import {Icon} from 'react-native-elements/dist/icons/Icon';
import Animated, {
  Extrapolate,
  interpolate,
  interpolateColor,
  useAnimatedStyle,
  useSharedValue,
  withSpring,
} from 'react-native-reanimated';
import {
  NotificationColor,
  NotificationIcon,
  NotificationNumber,
  NotificationType,
} from '../utils/types';

const TIMEOUT = 3000;

const Notifications = React.forwardRef((_, ref: any) => {
  const insets = useSafeAreaInsets();
  const [text, setText] = useState('');
  const [type, setType] = useState<NotificationType>('info');
  const [prevType, setPrevType] = useState<NotificationType>('info');

  const [height, setHeight] = useState(Dimensions.get('screen').height);
  const timerRef = useRef<any | null>(null);

  const isOpened = useSharedValue(0);
  const colorNow = useSharedValue(NotificationNumber[type]);

  const rStyles = useAnimatedStyle(() => ({
    backgroundColor: interpolateColor(
      colorNow.value,
      [NotificationNumber[prevType], NotificationNumber[type]],
      [NotificationColor[prevType], NotificationColor[type]],
    ),
    transform: [
      {
        translateY: interpolate(
          isOpened.value,
          [0, 1],
          [0, height],
          Extrapolate.CLAMP,
        ),
      },
    ],
  }));

  const notificationStyles = {
    backgroundColor: NotificationColor[type],
    paddingTop: insets.top + 21,
  };

  useImperativeHandle(ref, () => ({
    stack: [],
    show: (msg: string, t: NotificationType = 'info') => {
      if (!msg || msg === '') {
        return;
      } else {
        ref.current?.hide();

        isOpened.value = withSpring(1, springConfig);

        setText(msg);
        setPrevType(NotificationNumber[colorNow.value] as NotificationType);
        setType(t);

        colorNow.value = withSpring(NotificationNumber[t], springConfig);
        timerRef.current = setTimeout(ref.current?.hide, TIMEOUT);
      }
    },
    hide: () => {
      isOpened.value = withSpring(0, springConfig);

      clearTimeout(timerRef.current);
    },
  }));

  return (
    <Animated.View
      style={[styles.container, {top: -height}, notificationStyles, rStyles]}
      onLayout={e => setHeight(e.nativeEvent.layout.height)}>
      <TouchableOpacity onPress={ref.current?.hide}>
        <View style={styles.inner}>
          <Icon
            name={NotificationIcon[type]}
            type={'ionicon'}
            size={28}
            color={colors.white}
          />
          <View style={styles.textContainer}>
            <Text style={styles.text}>{text}</Text>
          </View>
        </View>
      </TouchableOpacity>
    </Animated.View>
  );
});

const styles = StyleSheet.create({
  container: {
    position: 'absolute',
    top: 0,
    left: 0,
  },
  inner: {
    width: Dimensions.get('screen').width,
    paddingHorizontal: 21,
    paddingBottom: 14,
    flexDirection: 'row',
    justifyContent: 'flex-start',
    alignItems: 'center',
  },
  textContainer: {
    paddingLeft: 7,
    flex: 1,
  },
  text: {
    fontSize: 16,
    fontWeight: '500',
    color: colors.white,
    flexShrink: 1,
  },
});

export default Notifications;
