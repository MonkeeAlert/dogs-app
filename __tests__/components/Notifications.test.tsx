import React, {Ref, RefObject} from 'react';
import renderer, {act} from 'react-test-renderer';
import {render, fireEvent} from '@testing-library/react-native';
import {shallow} from 'enzyme';
import Notifications from '../../src/components/Notifications';
import {notificationRef} from '../../src/utils/constants';
import * as ReactNative from 'react-native';

describe('Notifications', () => {
  it('matches snapshot', () => {
    act(() => {
      const tree = renderer
        .create(<Notifications ref={notificationRef} />)
        .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  it('does not show notification with empty message', () => {
    act(() => {
      const spy = jest.spyOn(notificationRef.current, 'show');
      const isVisible = notificationRef.current?.show('', 'info');

      expect(spy).toHaveBeenCalled();
      expect(isVisible).toBe(false);
    });
  });

  it('shows notification with non-empty message and default type', () => {
    act(() => {
      const spy = jest.spyOn(notificationRef.current, 'show');
      const isVisible = notificationRef.current?.show('info');

      expect(spy).toHaveBeenCalled();
      expect(isVisible).toBe(true);
    });
  });

  it('shows notification with non-empty message and custom type', () => {
    act(() => {
      const spy = jest.spyOn(notificationRef.current, 'show');
      const isVisible = notificationRef.current?.show('error', 'error');

      expect(spy).toHaveBeenCalled();
      expect(isVisible).toBe(true);
    });
  });

  it('renders and triggers onLayout', () => {
    const {getByTestId} = render(<Notifications ref={notificationRef} />);
    const view = getByTestId('Notification');

    act(() => {
      const evt = fireEvent(view, 'onLayout', {
        nativeEvent: {
          layout: {
            height: 0,
          },
        },
      });

      expect(evt).toBeUndefined();
    });
  });
});
