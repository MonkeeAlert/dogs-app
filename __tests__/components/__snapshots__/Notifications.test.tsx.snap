// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notifications matches snapshot 1`] = `null`;
